Subject: Make source buildable on Debian

The fixes/* directory shouldn't contain any Debian-specific fixes, only common
ones. Unfortunately, this approach fails without patching the Makefile, simply
because the original source (cron 3.0) was release in 1993, and certain things
such as POSIX are considered a "feature".
Index: patched/Makefile
===================================================================
--- patched.orig/Makefile	2010-05-06 18:18:16.350561149 +0200
+++ patched/Makefile	2010-05-06 18:18:21.821436206 +0200
@@ -57,28 +57,32 @@
 #<<need getopt()>>
 LIBS		=
 #<<optimize or debug?>>
-#OPTIM		=	-O
-OPTIM		=	-g
+OPTIM		=	-O2
+#OPTIM		=	-g
 #<<ATT or BSD or POSIX?>>
 # (ATT untested)
 #COMPAT		=	-DATT
 #(BSD is only needed if <sys/params.h> does not define it, as on ULTRIX)
 #COMPAT		=	-DBSD
 # (POSIX)
-#COMPAT		=	-DPOSIX
+COMPAT		=	-DPOSIX
 #<<lint flags of choice?>>
 LINTFLAGS	=	-hbxa $(INCLUDE) $(COMPAT) $(DEBUGGING)
 #<<want to use a nonstandard CC?>>
 #CC		=	vcc
 #<<manifest defines>>
 DEFS		=
+# The -DUSE_SIGCHLD is needed for the Alpha port
+DEFS = -DDEBIAN -DUSE_SIGCHLD
 #(SGI IRIX systems need this)
 #DEFS		=	-D_BSD_SIGNALS -Dconst=
 #<<the name of the BSD-like install program>>
 #INSTALL = installbsd
-INSTALL = install
+INSTALL = install -s
 #<<any special load flags>>
-LDFLAGS		=
+# LDFLAGS		=	-s
+# Let install do the strip
+
 #################################### end configurable stuff
 
 SHELL		=	/bin/sh
@@ -113,13 +117,14 @@
 			$(CC) $(LDFLAGS) -o crontab $(CRONTAB_OBJ) $(LIBS)
 
 install		:	all
-			$(INSTALL) -c -m  111 -o root -s cron    $(DESTSBIN)/
-			$(INSTALL) -c -m 4111 -o root -s crontab $(DESTBIN)/
+			$(INSTALL) -c -m  755 -o root cron    $(DESTSBIN)/
+			$(INSTALL) -c -m 4755 -o root crontab $(DESTBIN)/
 			sh putman.sh crontab.1 $(DESTMAN)
 			sh putman.sh cron.8    $(DESTMAN)
 			sh putman.sh crontab.5 $(DESTMAN)
 
-clean		:;	rm -f *.o cron crontab a.out core tags *~ #*
+clean		:
+			rm -f *.o cron crontab a.out core tags *~ #*
 
 kit		:	$(SHAR_SOURCE)
 			makekit -m -s99k $(SHAR_SOURCE)
